shell/platform/darwin/graphics/FlutterDarwinContextMetalImpeller.mm

```diff
@@ -42,7 +45,8 @@ @implementation FlutterDarwinContextMetalImpeller
 - (instancetype)init {
   self = [super init];
   if (self != nil) {
-    _context = CreateImpellerContext();
+    _workers = fml::ConcurrentMessageLoop::Create();
```

Comment:
We shouldn't create a message loop here. Instead, we should be getting the concurrent loop the engine creates already and passing it in as a parameter here.

That means we should mark the default initializer as unavailable and make a new one like `initWithMessageLoop`, or alternatively just `initWithTaskRunner` since that's all we actually need here.

Reply:
I'm not sure exactly how to do this. It looks like ImpellerContext is setup without any references to the current engine. Is this something we can/should change, or should I look at providing the message loop some other way?

Reply:
I think it'd flow through `PlatformViewIOS (ctor)` -> `IOSContext::Create` -> `IOSContextMetalImpeller (ctor)`

Looks like there's something left to figure out there though and we're _not_ doing this on Android right now (we create a special concurrent loop for the context to use there in [`android_context_vulkan_impeller.cc`](https://github.com/flutter/engine/blob/6bc60c8a98775f9b11dfa7ce7baf4a6665b3f166/shell/platform/android/android_context_vulkan_impeller.cc#L44)

Maybe for now we can file a bug for this. We shouldn't really be creating multiple concurrent message loops. But there might need to be some refactoring in the shell/platformview/engine setup to make sure that ownership and sharing of the loop is handled properly.

Reply:
Filed https://github.com/flutter/flutter/issues/127160. I'm ambivalent about whether resolving that blocks this or not, but if it turns out to be not too hard to resolve that then we should do it before creating another violating of it here.

Reply:
I think we should figure out the concurrent loop stuff first.